board/common/rootfs/usr/share/udhcpc/default.script

```diff
@@ -0,0 +1,134 @@
+#!/bin/sh
+
+# udhcpc script edited by Tim Riker <Tim@Rikers.org>
+
+[ -z "$1" ] && echo "Error: should be called from udhcpc" && exit 1
+
+ACTION="$1"
+RESOLV_CONF="/etc/resolv.conf"
+[ -e $RESOLV_CONF ] || touch $RESOLV_CONF
+[ -n "$broadcast" ] && BROADCAST="broadcast $broadcast"
+[ -n "$subnet" ] && NETMASK="$subnet"
+# Handle stateful DHCPv6 like DHCPv4
+[ -n "$ipv6" ] && ip="$ipv6/128"
+
+if [ -z "${IF_WAIT_DELAY}" ]; then
+	IF_WAIT_DELAY=10
+fi
+
+wait_for_ipv6_default_route() {
+	printf "Waiting for IPv6 default route to appear"
+	while [ $IF_WAIT_DELAY -gt 0 ]; do
+		if ip -6 route list | grep -q default; then
+			printf "\n"
+			return
+		fi
+		sleep 1
+		printf "."
+		: $((IF_WAIT_DELAY -= 1))
+	done
+	printf " timeout!\n"
+}
+
+flush_dhcp_addresses() {
+	addrs=$(ip -j addr show dev $interface | jq -c \
+		'.[0].addr_info[] | select(.family == "inet") | select(.protocol == "dhcp")')
+
+	for addr in $addrs; do
+		ip="$(echo "$addr" | jq -r '."local"')"
+		prefix="$(echo "$addr" | jq -r '."prefixlen"')"
+		ip addr del "$ip/$prefix" dev "$interface"
+	done
+}
+
+case "$ACTION" in
+	deconfig)
+		flush_dhcp_addresses
+
+		/bin/ip link set dev $interface up
+
+		# drop info from this interface
+		# resolv.conf may be a symlink to /tmp/, so take care
+		TMPFILE=$(mktemp)
+		grep -vE "# $interface\$" $RESOLV_CONF > $TMPFILE
+		cat $TMPFILE > $RESOLV_CONF
+		rm -f $TMPFILE
+
+		if [ -x /usr/sbin/avahi-autoipd ]; then
+			/usr/sbin/avahi-autoipd -c $interface && /usr/sbin/avahi-autoipd -k $interface
+		fi
+		;;
+
+	leasefail|nak)
+		if [ -x /usr/sbin/avahi-autoipd ]; then
+			/usr/sbin/avahi-autoipd -c $interface || /usr/sbin/avahi-autoipd -wD $interface --no-chroot
+		fi
+		;;
+
+	renew|bound)
+		if [ -x /usr/sbin/avahi-autoipd ]; then
+			/usr/sbin/avahi-autoipd -c $interface && /usr/sbin/avahi-autoipd -k $interface
+		fi
+		/bin/ip addr add dev $interface $ip/$NETMASK $BROADCAST proto 5
```

Comment:
For `proto 5`, and `proto 4` below is going to become pretty nasty.  I thought it would be possible to use the names from rt_addrprotos instead?

Reply:
Strangely the upstream code doesn't allow ASCII when setting proto.
The manual states `ADDRPROTO := [ NAME | NUMBER ]` but adding a string throws `Error: argument "static" is wrong: "proto" value is invalid`. Checking the code it does:
```                                                                                                                                                     
172 +  if (get_u8(&proto, *argv, 0))                                                                                                                                        
173 +      invarg("\"proto\" value is invalid\n", *argv);
```

Maybe this is fixed in later upstream patches. The number does however resolve to ASCII during printing (`ip addr show`). I think we can fix this in a later patchset?

Reply:
Aha, yeah must be a missing patch upstream.  Right, let's fix that later!